cpp1st/week06/taehwan/README.md

```diff
@@ -0,0 +1,448 @@
+# Type Deduction
+
+## (```auto```-related) Template Type Deduction
+General problem:
+
+```c++
+template<typename T>
+void f(ParamType param);
+f(expr);        // deduce T and ParamType from expr
```

Comment:
ParamType must be T or T's reference, pointer. Right?

Reply:
Yes, right! And sometimes ParamType can be combined with keyward const, volatile and so on.